lib/ansible/module_utils/network/vyos/config/lag_interfaces/lag_interfaces.py

```diff
@@ -0,0 +1,334 @@
+        warnings = list()
+        existing_lag_interfaces_facts = self.get_lag_interfaces_facts()
+        commands.extend(self.set_config(existing_lag_interfaces_facts))
+        if commands:
+            if self._module.check_mode:
+                resp = self._connection.edit_config(commands, commit=False)
+            else:
+                resp = self._connection.edit_config(commands)
+            result['changed'] = True
+
+        result['commands'] = commands
+
+        if self._module._diff:
+            result['diff'] = resp['diff'] if result['changed'] else None
+
+        changed_lag_interfaces_facts = self.get_lag_interfaces_facts()
+
+        result['before'] = existing_lag_interfaces_facts
+        if result['changed']:
+            result['after'] = changed_lag_interfaces_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_lag_interfaces_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        have = existing_lag_interfaces_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        state = self._module.params['state']
+        if state == 'overridden':
+            commands.extend(self._state_overridden(want=want, have=have))
+        elif state == 'deleted':
+            if want:
+                for item in want:
+                    name = item['name']
+                    obj_in_have = search_obj_in_list(name, have)
+                    commands.extend(self._state_deleted(have_lag=obj_in_have))
+            else:
+                for item in have:
+                    commands.extend(self._state_deleted(have_lag=item))
+        elif state == 'merged':
+            for item in want:
+                name = item['name']
+                obj_in_have = search_obj_in_list(name, have)
+                commands.extend(self._state_merged(want_lag=item, have_lag=obj_in_have))
+        elif state == 'replaced':
+            for item in want:
+                name = item['name']
+                obj_in_have = search_obj_in_list(name, have)
+                commands.extend(self._state_replaced(want_lag=item, have_lag=obj_in_have))
+        return commands
+
+    @staticmethod
+    def _state_replaced(**kwargs):
```

Comment:
Using kwargs is probably redundant cause we will always be passing the same arguments to the _state_* methods.